cmd/clusterctl/pkg/client/cluster/upgrader_test.go

```diff
@@ -428,6 +428,55 @@ func Test_providerUpgrader_Plan(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "Single Management group, no multi-tenancy, upgrade for two contracts, but the upgrade for the second one is partially available",
+			fields: fields{
+				// config for two providers
+				reader: test.NewFakeReader().
+					WithProvider("core", clusterctlv1.CoreProviderType, "https://somewhere.com").
```

Comment:
From the CI test failure, I believe changing references of `core` to `cluster-api` might fix it.
```suggestion
					WithProvider("cluster-api", clusterctlv1.CoreProviderType, "https://somewhere.com").
```

Reply:
I'm not sure if CI merges the PR with master and then runs the tests or if it just runs the tests on the same branch.

Reply:
good catch!
fixed